### PR TITLE
Kh questionnaire mobile support

### PR DESCRIPTION
**Pull request by kurvinh**
questionnaire @ bff2cfc on stage-moodle35 where are at this commit here
which you can see below, and all the updates for moodle35 that are not on the forked version of master. 
we are currently on bff2cfc on https://github.com/getsmarter/online-campus/tree/master/mod

### CODE REVIEW

File: classes/output/mobile.php [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        // Capabilities check.
+        $cm = get_coursemodule_from_id('questionnaire', $cmid);
+        $context = \context_module::instance($cmid);
+        self::require_capability($cm, $context, 'mod/questionnaire:view');
+        // Set some variables we are going to be using.
+        $questionnaire = get_questionnaire_data($cmid, $USER->id);
+        if (isset($questionnaire['questions'][$pagenum-1]) && !empty($questionnaire['questions'][$pagenum-1])) {
+            $prevpage = $pagenum-1;
+        }
+        //$jsfilepath = $CFG->wwwroot . '/mod/questionnaire/javascript/mobile.js';
```
Comment (JJSwanevelder) [inline classes/output/mobile.php]: Any reason why this is commented out

File: classes/output/mobile.php [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+                $DB->get_record('course', ['id' => $cm->course]), $cm);
+            $rid = $DB->get_field('questionnaire_response', 'id',
+                [
+                    'questionnaireid' => $questionnaire['questionnaire']['questionnaireid'],
+                    'complete' => 'n',
+                    'userid' => $USER->id
+                ]);
+            if (isset($questionnaire['questions'][$pagenum]) && !empty($questionnaire['questions'][$pagenum])) {
+                // Maybe for future
+                //$qnumplus = 0;
```
Comment (JJSwanevelder) [inline classes/output/mobile.php]: Do we need to keep this commented out code?

File: classes/output/mobile.php [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+                // Search for the next page to output
+                while (!$questionnaireobj->eligible_questions_on_page($pagenum, $rid)) {
+                    if (isset($questionnaire['questions'][$pagenum]) && !empty($questionnaire['questions'][$pagenum])) {
+                        /*if ($questionnaire['questionnaire']['autonumquestions']) {
```
Comment (JJSwanevelder) [inline classes/output/mobile.php]: Same here

File: classes/output/mobile.php (hunk 1 above)
Comment (kurvinh) [reply classes/output/mobile.php]: I can actually remove that, it's just a debug thanks

File: db/mobile.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                'displaydata' => [
+                    'icon' => $CFG->wwwroot . '/mod/questionnaire/pix/icon.gif',
+                    'class' => '',
+                ],
+                'delegate' => 'CoreCourseModuleDelegate',
+                'method' => 'mobile_view_activity'
+            ]
+        ],
+        'lang' => [
+            ['yourresponse', 'questionnaire'],
```
Comment (JJSwanevelder) [inline db/mobile.php]: Just some advice regarding language files. This array can get messy if the the required strings get out of hand. I have moved most of my language strings to the mobile.php classes and then tying into moodle's default get_string functionality then pass those strings to the template.

Comment (kurvinh) [reply db/mobile.php]: ah yeah rings a bell, I have seen that in legend themes will update.

**Pull request merged**